### ISSUE TITLE
'Successfully synchronized your repository list' even though Gitea's logs show 401 Unauthorised

### ISSUE THREAD

**Issue by reivilibre** [woodpecker-ci/woodpecker#503]
### Component

server, web-ui

### Describe the bug

One of my new repositories was not showing up in the repository list.
I clicked 'Synchronise' in the web UI and there was a notification telling me that it was successful.
My new repository did not show up. I didn't see anything in Woodpecker's logs and I spent a long time looking through the code to figure it out.
Eventually I looked through the Gitea logs and saw that the request failed with a 401 Unauthorised error.

If I recall, logging out through the web UI and then logging in again caused the synchronise button to work again.

(There might be a separate issue here about Gitea access tokens expiring or something? But ideally a Gitea API error should not show up as successful. Oops!)

### System Info

```shell
{"source":"https://github.com/woodpecker-ci/woodpecker","version":"${VERSION}"}
```
... hmm.

I'm using the `florianpiesche/woodpecker-server:v0.14.1` Docker image on a RPi 4.
```


### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy).

**Comment by 6543** [comment]
that's an issue with the old UI, which is rewritten now and will be released with 0.15.0 - if you want to look into it try to fix it I'll merge that fix. but current efforts are moving towards release new code :)

**Comment by 6543** [comment]
call it a "wont fix" because: we focus on getting new version ready :)
